aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -192,7 +192,8 @@ private RESTClient httpClient() {
               HTTPClient.builder(properties())
                   .uri(baseSignerUri())
                   .withObjectMapper(S3ObjectMapper.mapper())
-                  .build();
+                  .build()
+                  .withAuthSession(org.apache.iceberg.rest.auth.AuthSession.EMPTY);
```

Comment:
maybe worth having a `withAuthSession` on the builder class itself(similar to `withObjectMapper`)

Reply:
Builder method added.